bot/seasons/christmas/adventofcode.py

```diff
@@ -0,0 +1,493 @@
+import json
+import logging
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import List
+
+import aiohttp
+import discord
+from bs4 import BeautifulSoup
+from discord.ext import commands
+
+from bot.constants import AdventOfCode as AocConfig
+from bot.constants import Colours, Emojis, Tokens
+
+log = logging.getLogger(__name__)
+
+AOC_REQUEST_HEADER = {"user-agent": "PythonDiscord AoC Event Bot"}
+AOC_SESSION_COOKIE = {"session": Tokens.aoc_session_cookie}
+
+
+class AdventOfCode:
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+        self._base_url = f"https://adventofcode.com/{AocConfig.year}"
+        self.global_leaderboard_url = f"https://adventofcode.com/{AocConfig.year}/leaderboard"
+        self.private_leaderboard_url = f"{self._base_url}/leaderboard/private/view/{AocConfig.leaderboard_id}"
+
+        self.about_aoc_filepath = Path("./bot/resources/advent_of_code/about.json")
+        self.cached_about_aoc = self._build_about_embed()
+
+        self.cached_global_leaderboard = None
+        self.cached_private_leaderboard = None
+
+    @commands.group(name="adventofcode", aliases=("aoc",), invoke_without_command=True)
+    async def adventofcode_group(self, ctx: commands.Context):
+        """
+        Advent of Code festivities! Ho Ho Ho!
+        """
+
+        await ctx.invoke(self.bot.get_command("help"), "adventofcode")
+
+    @adventofcode_group.command(name="about", aliases=("ab", "info"), brief="Learn about Advent of Code")
+    async def about_aoc(self, ctx: commands.Context):
+        """
+        Respond with an explanation all things Advent of Code
+        """
+
+        await ctx.send("", embed=self.cached_about_aoc)
+
+    @adventofcode_group.command(name="join", aliases=("j",), brief="Learn how to join PyDis' private AoC leaderboard")
+    async def join_leaderboard(self, ctx: commands.Context):
+        """
+        Reply with the link to join the PyDis AoC private leaderboard
```

Comment:
```suggestion
        Retrieve the link to join the PyDis AoC private leaderboard
```

Phrasing is a bit more favourable.